### PR TITLE
feat(group): make inital open state configurable

### PR DESCRIPTION
**Pull request by marstamm**
related to https://github.com/bpmn-io/bpmn-js-properties-panel/issues/618

for integration PR in bpmn-js properties panel, see https://github.com/bpmn-io/bpmn-js-properties-panel/pull/621

### CODE REVIEW

File: src/components/Group.js [hunk 1]
```diff
@@ -27,12 +27,13 @@ export default function Group(props) {
     element,
     entries = [],
     id,
-    label
+    label,
+    initiallyOpen = false,
```
Comment (pinussilvestrus) [inline src/components/Group.js:31]: Why not `shouldOpen` as we have in [other components](https://github.com/bpmn-io/properties-panel/blob/main/src/components/entries/Collapsible.js#L25)?

Comment (pinussilvestrus) [reply src/components/Group.js:31]: We should also make sure to update our [type docs](https://github.com/bpmn-io/properties-panel/blob/main/src/PropertiesPanel.js#L66) (although we don't validate with them by now).

**Comment by marstamm** [comment]
Ready for review. I also adjusted the downstream integration: https://github.com/bpmn-io/bpmn-js-properties-panel/pull/621

File: src/context/LayoutContext.js [hunk 2]
```diff
@@ -5,7 +5,7 @@ import {
 const LayoutContext = createContext({
   layout: {},
   setLayout: () => {},
-  getLayoutForKey: () => {},
+  getLayoutForKey: (key, defaultValue) => defaultValue,
```
Comment (pinussilvestrus) [inline src/context/LayoutContext.js:8]: Do we need to be that explicit at this stage?

**Review by pinussilvestrus — COMMENTED** [review]
Looks good in general, makes sense to make it possible to set the initial state from a parent component. 👍 

Just for my understanding: did we consider building it via the `useEvent` API [we recently introduced](https://github.com/bpmn-io/properties-panel/pull/137)? So to make it also possible to open groups from elsewhere? 

Probably that's a different use case because this `open` state should also be persisted, right?

File: src/context/LayoutContext.js (hunk 2 above)
Comment (marstamm) [reply src/context/LayoutContext.js]: You are right, we can mock it in the tests instead. I went ahead and did that

**Comment by marstamm** [comment]
> Just for my understanding: did we consider building it via the useEvent API https://github.com/bpmn-io/properties-panel/pull/137? So to make it also possible to open groups from elsewhere?

I did consider going this way. However, the use case is slightly different and not as clean as solving it via props. E.g., we would always have to fire the event right after rendering while also checking that the state is persisted. Also, the current API only supports elements and not groups.

In the end, I decided to go with `shouldOpen`, as we also have those 2 options side-by-side in the ListGroup component:
https://github.com/bpmn-io/properties-panel/blob/00c4b3be7c83927c77758d4ab6c2e58400e97436/src/components/ListGroup.js#L41

**Review by nikku — COMMENTED** [review]
I like the way this is going + agree to https://github.com/bpmn-io/properties-panel/pull/139#issuecomment-1077367615.

@pinussilvestrus Please give your :+1: to confirm implementation.

**Review by pinussilvestrus — APPROVED** [review]
Thanks for the clarification 👏  I agree 🙂

**Pull request merged**